packages/next/next-server/server/config.ts

```diff
@@ -83,10 +83,24 @@ function assignDefaults(userConfig: { [key: string]: any }) {
       experimentalWarning()
     }
 
-    if (key === 'distDir' && userConfig[key] === 'public') {
-      throw new Error(
-        `The 'public' directory is reserved in Next.js and can not be set as the 'distDir'. https://err.sh/zeit/next.js/can-not-output-to-public`
-      )
+    if (key === 'distDir') {
+      const userDistDir =
+        typeof userConfig[key] === 'string' ? userConfig[key].trim() : '.next'
+
+      // don't allow public as the distDir as this is a reserved folder for
+      // public files
+      if (userDistDir === 'public') {
+        throw new Error(
+          `The 'public' directory is reserved in Next.js and can not be set as the 'distDir'. https://err.sh/zeit/next.js/can-not-output-to-public`
+        )
+      }
+      // make sure distDir isn't an empty string which can result the provided
+      // directory being deleted in development mode
+      if (userDistDir.length === 0) {
```

Comment:
This will throw an error when users provide `undefined` or `null`.
```suggestion
      if (userDistDir?.length === 0) {
```

Can we add a test for this case?

Reply:
Added test case